packages/components/src/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { ReactElement, ReactNode } from "react";
+import { useAssert } from "@jobber/hooks/useAssert";
+import { ComboboxContent } from "./components/ComboboxContent";
+import { ComboboxAction } from "./components/ComboboxAction";
+import { ComboboxContextProvider } from "./ComboboxProvider";
+import {
+  ComboboxTriggerButton,
+  ComboboxTriggerChip,
+} from "./components/ComboboxTrigger";
+
+export interface ComboboxProps {
+  readonly children: ReactElement | ReactElement[];
+}
+
+export const COMBOBOX_TRIGGER_COUNT_ERROR_MESSAGE =
+  "Combobox can only have one Trigger element";
+export const COMBOBOX_REQUIRED_CHILDREN_ERROR_MESSAGE =
+  "Combobox must have a Trigger and Combobox.Content element";
+
+export const Combobox = (props: ComboboxProps): JSX.Element => {
+  const { contentElement, triggerElement } = useComboboxValidation(
+    props.children,
+  );
+  return (
+    <ComboboxContextProvider>
+      {triggerElement}
+      {contentElement}
+    </ComboboxContextProvider>
+  );
+};
+
+function useComboboxValidation(children: ReactNode): {
```

Comment:
good questions!

as I recall, the way we got here was more or less:

"it might be nice if combobox didn't care about the order you gave the subcomponents"

eg.

```
<Combobox>
  <Combobox.Trigger/>
  <Combobox.Content/>
</Combobox>
```
is just as valid as

```
<Combobox>
  <Combobox.Content/>
  <Combobox.Trigger/>
</Combobox>
```

to achieve that, we needed to identify which child was which. now that we know which child is which and how many we got, we thought why not add a bit of validation to improve the developer experience? 

there was also a 3rd thing we tried to get out of this setup that ended up not panning out. the goal of it was to avoid the use of a `Context` but yeah we didn't do it in the end

but you're right, this is more code that _could_ have bugs. is it worth it? debatable. I agree that if Atlantis makes (or has already made) a philosophical choice that says "if you do something silly, you're on your own" that can inform these implementation decisions, and it simplifies many of these discussions around how defensively we need to code/how much we need to protect people from themselves

maybe it's perfectly fine for our code to totally break if you give us two `Triggers` since that's something we've explicitly said we don't support!